server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1735,14 +1750,10 @@ public Boolean providesDerived(Pool pool, String derivedProvidedProductId) {
             }
 
             for (Object[] cols : (List<Object[]>) query.getResultList()) {
-                Set<String> providedProducts = providedProductMap.get((String) cols[0]);
-
-                if (providedProducts == null) {
-                    providedProducts = new HashSet<>();
-                    providedProductMap.put((String) cols[0], providedProducts);
-                }
-
-                providedProducts.add((String) cols[1]);
+                providedProductMap.computeIfAbsent((String) cols[0],
```

Comment:
Incorrect use of ```computeIfAbsent```. The method will generate a new value if not present *and* return whatever ends up in the field in either case. This saves us an interaction with the map by making the fetch, generation, and storage operations all atomic and packaged in a single call. You can then use method chaining and the knowledge that you'll never get a null value out of the map to eliminate the need for explicit declaration. It should look like this:

```
providedProductMap.computeIfAbsent((String) cols[0], key -> new HashSet<>())
    .add((String) cols[1]);
```

Reply:
Removed this piece of code from here.